src/Backend/Modules/Users/Actions/Edit.php

```diff
@@ -57,6 +57,26 @@ class Edit extends BackendBaseActionEdit
     public function execute()
     {
         $this->id = $this->getParameter('id', 'int');
+        $this->error = $this->getParameter('error', 'string');
+        $this->loadAuthenticatedUser();
+
+        // If id and error parameters are not set we'll assume the user logged in
+        // and has been redirected to this action by the authentication index action.
+        // When this is the case the user will be redirected to the index action of this module.
+        // An action to which he may not have any user rights.
+        // Redirect to the user's own profile instead to avoid unnessary words.
+        if(
+            $this->id === null &&
+            $this->error === null &&
+            $this->authenticatedUser->getUserId()
```

Comment:
@jeroendesloovere that's not really a rule in the coding styles. I like it more with the '&&' statements in the front, because it's easier to see tha it's a multiline if statement, but it is not required.